File: java/bob/src/main/java/Bob.java
public class Bob {
  String hey(String s) {
    if (s.chars().filter(c -> !Character.isWhitespace(c)).count() == 0) {
      return "Fine. Be that way!";
    }
    if (s.chars().filter(Character::isLetter).count() != 0
        && s.chars().filter(Character::isLetter).allMatch(Character::isUpperCase)) {
      return isQuestion(s) ? "Calm down, I know what I'm doing!" : "Whoa, chill out!";
    }
    if (isQuestion(s)) {
      return "Sure.";
    }

    return "Whatever.";
  }

  boolean isQuestion(String s) {
    return s.trim().endsWith("?");
  }
}

File: javascript/bob/bob.js
/* eslint-disable no-unused-vars */
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = message => {
  if (/^\s*$/.test(message)) {
    return "Fine. Be that way!";
  }

  const isQuestion = message.trimEnd().endsWith("?");

  if (/[A-Z]/.test(message) && message.toUpperCase() == message) {
    if (isQuestion) {
      return "Calm down, I know what I'm doing!";
    } else {
      return "Whoa, chill out!";
    }
  }

  if (isQuestion) {
    return "Sure.";
  }

  return "Whatever.";
};
